src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -270,23 +270,32 @@ internal unsafe static HttpVersion GetKnownVersion(byte* location, int length)
         /// <param name="knownScheme">A reference to the known scheme, if the input matches any</param>
         /// <returns>True when memory starts with known http or https schema</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
+        public static unsafe bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
         {
-            if (span.TryRead<ulong>(out var value))
+            fixed (byte* data = &span.DangerousGetPinnableReference())
```

Comment:
ok, i don't know enough about what corefx is actually planning to release so this is fine for now.